perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/VosManagerMethod.java

```diff
@@ -482,20 +498,32 @@ public Void call(ApiCaller ac, Deserializer parms)
 	getAdmins {
 		@Override
 		public List<User> call(ApiCaller ac, Deserializer parms) throws PerunException {
-			if(parms.contains("role")) {
-				String roleName = parms.readString("role");
-				if (roleName == null) {
-					throw new RpcException(RpcException.Type.WRONG_PARAMETER, "Parameter role cannot be null.");
+			try {
+				if(parms.contains("role")) {
+					String roleName = parms.readString("role");
+					if (roleName == null) {
+						throw new RpcException(RpcException.Type.WRONG_PARAMETER, "Parameter role cannot be null.");
+					} else {
+						roleName = roleName.toUpperCase();
+					}
+
+					if (!AuthzResolver.roleExists(roleName)) {
+						throw new RoleNotSupportedException("Role: "+ roleName +" does not exists.", roleName);
+					}
+
+					//Role can be only supported one (TopGroupCreator, VoAdmin or VoObserver)
+					if(!roleName.equals(Role.TOPGROUPCREATOR) &&
+						!roleName.equals(Role.VOADMIN) &&
+						!roleName.equals(Role.VOOBSERVER)) {
+						throw new RoleNotSupportedException("Supported roles are VoAdmin, VoObserver and TopGroupCreator.", roleName);
+					}
```

Comment:
I think something similar `getRichAdmins` already does with `objectAndRoleManageableByEntity`, but if we want to exactly preserve the old behaviour, this is fine

Reply:
Yes, the behavior of methods should stay exactly the same. Therefore, the same kind of exception should be thrown in case of an incorrect input role.